initial documentation of `TaskSignal.any` (#30314)
mdn/content: files/en-us/web/api/tasksignal/any_static/index.md

---
title: "TaskSignal: any() static method"
short-title: any()
slug: Web/API/TaskSignal/any_static
page-type: web-api-static-method
status:
  - experimental
browser-compat: api.TaskSignal.any_static
---

{{APIRef("Prioritized Task Scheduling API")}}{{SeeCompatTable}}

> The **`TaskSignal.any()`** static method takes an iterable of abort signals and returns a {{domxref("TaskSignal")}}. The returned task signal is aborted when any of the abort signals is aborted. The {{domxref("AbortSignal.reason", "reason")}} property will be set to the reason of the first signal that is aborted. The {{domxref("TaskSignal.priority", "priority")}} property will be determined by the specific `priority` parameter, which can either be a a string which is one of `user-blocking`, `user-visible` and `background` or a `TaskSignal`, in which case the new signal's `priority` will change along with this signal.

Reviewer: ```suggestion
The **`TaskSignal.any()`** static method takes an iterable of {{domxref("AbortSignal")}} objects and returns a {{domxref("TaskSignal")}}. The returned task signal is aborted when any of the abort signals is aborted.

When the task signal is aborted, its {{domxref("AbortSignal.reason", "reason")}} property will be set to the reason of the first signal that is aborted.
```

(we should talk about `priority` below, don't need to cover it here as well)

## Syntax

```js-nolint
TaskSignal.any(signals)
TaskSignal.any(signals, init)
```

### Parameters

- `signals`
  - : An [iterable](/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol) (such as an {{jsxref("Array")}}) of abort signals.
- `init` {{optional_inline}}
  - : Contains optional configuration parameters. Currently only one property is defined:
    - `priority` {{optional_inline}}

>       - : A string which is one of `user-blocking`, `user-visible` and `background`, or an {{domxref("AbortSignal")}} (since {{domxref("TaskSignal")}} inherits from {{domxref("AbortSignal")}}, so a {{domxref("TaskSignal")}} can also be accepted).

Reviewer: ```suggestion
      - : One of the following:
        - A string which is one of `user-blocking`, `user-visible` and `background`.
        - A {{domxref("TaskSignal")}}.
```

(I think it can't be an `AbortSignal` (https://wicg.github.io/scheduling-apis/#dictdef-tasksignalanyinit) which makes sense because `AbortSignal` doesn't have a `priority`.)

Author: yes, my mistake here. the `priority` can not accept a `AbortSignal`

### Return value

> A `TaskSignal` instance, its {{domxref("AbortSignal.reason", "reason")}} property will be set to the reason of the first signal that is aborted, and its {{domxref("TaskSignal.priority", "priority")}} property will be determined by the specific `priority` parameter.

Reviewer: ```suggestion
A `TaskSignal` instance. It will be aborted when the first signal passed into `signals` is aborted. When this happens:

- Its {{domxref("AbortSignal.reason", "reason")}} property will be set to the reason of the signal that caused this signal to abort.

- Its {{domxref("TaskSignal.priority", "priority")}} property will be determined by the `priority` parameter:

  - If the `priority` parameter was a string, it will be the value of the string.
  - If the `priority` parameter was a `TaskSignal`, it will be the value of that signal's `priority`.
```
